delivery/client.py

```diff
@@ -179,14 +180,24 @@ def upload_metadata(
         self,
         data: typing.Iterable[dso.model.ArtefactMetadata],
     ):
-        res = self.session.post(
-            url=self._routes.upload_metadata(),
+        headers = {
+            'Content-Type': 'application/json',
+        }
```

Comment:
nit: you might offload setting content-type to encode_request. otoh: explicit is better than implicit.